### ISSUE TITLE
ENSEMBL transcripts not versioned

### ISSUE THREAD

**Issue by ahwagner** [biocommons/uta#233]
When trying to refer to ensembl transcripts we cannot find by version in the 20210129 data release.

**Comment by reece** [comment]
This issue touches on several problems:
* The Ensembl data are extremely old.
* At the time of last loading, Ensembl didn't expose transcript versions. They eventually did so.
* It's extremely hard to get exon structures out of Ensembl.

The only official way to get exon coordinates out of Ensembl is to use the perl API. Unfortunately, when I last tried in May 2016, I discovered that Ensembl and bioperl didn't work on a modern distribution.

In ensembl-dev:
> Hi Reece,
> 
> Yes, you are right. It's the perl version that's not compatible with BioPerl and even the Ensembl API. The latest perl version that's been safely tested with Ensembl API is 5.14.
> 
> Thanks,
> Harpreet

My recollection is that perl 5.14 was significantly out of date at the time, and that installing manually had knock-on effects with dependent modules. I gave up.

So, to solve this issue, we need a reliable way to get versioned transcripts out of Ensembl.

**Comment by ahwagner** [comment]
Would we be able to use the .gff files for this purpose, e.g. http://ftp.ensembl.org/pub/release-104/gff3/homo_sapiens/Homo_sapiens.GRCh38.104.chr_patch_hapl_scaff.gff3.gz? It appears that they have gene/transcript/exon IDs with versions, and earlier releases are also maintained, e.g. release 101: http://ftp.ensembl.org/pub/release-101/gff3/homo_sapiens/Homo_sapiens.GRCh38.101.chr_patch_hapl_scaff.gff3.gz

**Comment by reece** [comment]
Yes, those should be usable in principle, but no work has actually gone into that yet.

**Comment by davmlaw** [comment]
Hi, I've made [cdot - data provider](https://github.com/SACGF/cdot) that includes Ensembl transcripts - see [HGVS issue](https://github.com/biocommons/hgvs/issues/634)

**Comment by jsstevenson** [comment]
I think our intention (hope?) was to keep this open, given @ahwagner 's tag on 2023-11-28 -- not sure why the stalebot closed it anyway?

**Comment by ahwagner** [comment]
Steps needed:
1.  Establish a base set of sequences for Ensembl (loaded from SeqRepo)
2. Write methods to check for changes to sequence digest with shared Ensembl ID
3. Version Ensembl IDs in UTA if needed
4. Expose version info in SeqRepo REST service service info
5. Build and deploy to UTA and SeqRepo psql database
